core_java/jdbc/src/main/java/ca/jrvs/apps/jdbc/CustomerDAO.java

```diff
@@ -0,0 +1,188 @@
+package ca.jrvs.apps.jdbc;
+
+import ca.jrvs.apps.jdbc.util.DataAccessObject;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CustomerDAO extends DataAccessObject<Customer> {
+  //SQL statement for functions
+  private static final String INSERT = "INSERT INTO customer (first_name, last_name," +
+      "email, phone, address, city, state, zipcode) VALUES (?, ?, ?, ?, ?, ?, ?, ?)";
+
+  private static final String GET_ONE = "SELECT customer_id, first_name, last_name, " +
+      "email, phone, address, city, state, zipcode FROM customer WHERE customer_id=?";
+
+  public CustomerDAO(Connection connection) {
```

Comment:
It's suggested to put member fields together, move this constructor down after member fields. Please remove the extra empty lines.